R/collapse_counts.R

```diff
@@ -1,23 +1,71 @@
+#' validate_columns_exist
+#' 
+#' This function checks that a list of columns are present in a dataframe.
+#' 
+#' @param selected_columns A vector of strings each representing a column name
+#' @param df A dataframe to check against
+#' @return Boolean
+validate_columns_exist= function(selected_columns, df) {
+  # Check that all of selected_columns are in df
+  if(any(!selected_columns %in% colnames(df))) {
+    return(FALSE)
+  } else {
+    return(TRUE)
+  }
+}
+
+#' validate_num_bio_reps
+#' 
+#' This function checks that all the expected flowcells are present in a table of detected flowcells.
+#' There can be more detected flowcells than there are expected flowcells.
+#' 
+#' @param detected_flowcells A dataframe with the columns "flowcell_name" and "flowcell_lane".
+#' @param expected_flowcells A dataframe with the columns "flowcell_name" and "flowcell_lane".
+validate_num_bio_reps= function(max_bio_rep_id, max_bio_rep_count) {
+  if(max_bio_rep_id > 1 & max_bio_rep_count == 1) {
+    stop('Unable to collapse bio_reps over the specified sig_cols.')
+  } else if(max_bio_rep_id < max_bio_rep_count) {
+    stop('Bio_reps were incorrectly collapsed resulting in more replicates than specified.')
+  } else if(max_bio_rep_id > 1 & max_bio_rep_id > max_bio_rep_count) {
+    print('Warning - Number of replicates that were collapses is smaller than the expected number of replicates.')
+    print('This could be due to a problem in processing or from poor data/sequencing quality.')
+  } else {}
+}
+
 #' collapse_counts
 #' 
 #' collapses l2fc values and computes MAD/sqrt(n) metrics for treatment conditions
 #'
 #'  @param l2fc - l2fc table with MAD/sqrt(n) metric for control condition
 #'  @return - collapsed_counts 
 #'  @export 
-collapse_counts = function(l2fc) {
+collapse_counts = function(l2fc, sig_cols) {
+  # Potentially static columns?
+  static_cols= c('project_code', 'CCLE_name', 'DepMap_ID', 'sig_id')
+  
+  # Validation: Check that sig_cols are present in l2fc.
+  if(validate_columns_exist== FALSE) {
```

Comment:
seems like you are checking if the function equates to false rather than calling the function and then checking if the result is false?